### PR TITLE
Adding `EraDuration`  function

### PR DESCRIPTION
**Pull request by Jay-sanjay**
**This PR is related to #5**

## PR CheckList

- [x] Added the very first dispatch for `EraDuration` function
- [x] Added the test-suite for the same
- [x]  Added the Doc-strings  
- [x]  Added the remaining dispatches with test-suite

### CODE REVIEW

**Comment by TheCedarPrince** [comment]
Just a note that the original implementation comes from this line of code: https://github.com/mi-erasmusmc/TreatmentPatterns/blob/171b60af7e76f63e09f2c2d48e2b811ce1dafff0/R/ConstructPathways.R#L195C1-L205C2 -- still doing a review here! :smile:

File: src/preprocessing.jl [hunk 1]
```diff
@@ -26,4 +26,23 @@ function Dummy(
...
+Given a treatment history dataframe, this function filters out rows where the difference between drug_exposure_start and drug_exposure_end is less than minEraDuration.
+
+    # Arguments:
+
+    - treatment_history: DataFrame, treatment history dataframe.
+    - minEraDuration: Int, minimum duration of an era.
+
+    # Returns:
+    - Updated dataframe, rows where the difference between drug_exposure_start and drug_exposure_end is less than minEraDuration are filtered out.
+
```
Comment (TheCedarPrince) [inline src/preprocessing.jl:46]: Hey Jay, could you add an example here?

File: src/preprocessing.jl [hunk 2]
```diff
@@ -26,4 +26,23 @@ function Dummy(
 
 end
 
-export Dummy
+"""
+Given a treatment history dataframe, this function filters out rows where the difference between drug_exposure_start and drug_exposure_end is less than minEraDuration.
```
Comment (TheCedarPrince) [inline src/preprocessing.jl]: Could you format the docstring to look more like this, Jay: https://github.com/JuliaHealth/OMOPCDMCohortCreator.jl/blob/637960b01f08ec3ce6e66c73d74bddf8556dde2e/src/getters.jl#L420 ? With proper markup and everything?

File: src/preprocessing.jl [hunk 3]
```diff
@@ -26,4 +26,23 @@ function Dummy(
...
+    - treatment_history: DataFrame, treatment history dataframe.
+    - minEraDuration: Int, minimum duration of an era.
+
+    # Returns:
+    - Updated dataframe, rows where the difference between drug_exposure_start and drug_exposure_end is less than minEraDuration are filtered out.
+
+"""
+function EraDuration(treatment_history::DataFrame, minEraDuration)
+    treatment_history = filter(row -> (row[:drug_exposure_end] - row[:drug_exposure_start]) >= minEraDuration, treatment_history)
+    @info "After minEraDuration: $(nrow(treatment_history))"  # For debugging purposes
```
Comment (TheCedarPrince) [inline src/preprocessing.jl]: Could you clean up this function and remove debugging?

File: src/preprocessing.jl [hunk 4]
```diff
@@ -26,4 +26,23 @@ function Dummy(
...
+    # Arguments:
+
+    - treatment_history: DataFrame, treatment history dataframe.
+    - minEraDuration: Int, minimum duration of an era.
+
+    # Returns:
+    - Updated dataframe, rows where the difference between drug_exposure_start and drug_exposure_end is less than minEraDuration are filtered out.
+
+"""
+function EraDuration(treatment_history::DataFrame, minEraDuration)
```
Comment (TheCedarPrince) [inline src/preprocessing.jl]: Could we rename the function to `calculate_era_duration`?

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Just some small tweaks -- great job Jay!

**Review by TheCedarPrince — APPROVED** [review]
LGTM -- Let's merge!

**Comment by Jay-sanjay** [comment]
@TheCedarPrince  the reason that deployment was failing was that it was not included to the api.md files. Though I have added it now it might throw a merge conflict because we have a same thing added in `period_prior_to_index`

**Review by TheCedarPrince — APPROVED** [review]
Ah thanks for the fixes! Let's merge it in! :smile: Go ahead and merge! :D

**Review by TheCedarPrince — APPROVED** [review]
Ah thanks for the fixes! Let's merge it in! :smile: Go ahead and merge! :D

**Pull request merged**